spec/models/supplier_number_spec.rb

```diff
@@ -44,37 +33,49 @@
       expect(supplier).to be_valid
     end
 
-    it 'is invalid if postcode is filled and has the wrong format' do
-      supplier.postcode = 'not-a-valid-postcode'
-      expect(supplier).not_to be_valid
-      expect(supplier.errors[:postcode]).to include('Enter a valid postcode')
+    context 'with wrong format' do
+      before do
+        supplier.postcode = 'not-a-valid-postcode'
+        supplier.validate
+      end
+
+      it { expect(supplier).to be_invalid }
+      it { expect(supplier.errors[:postcode]).to include('Enter a valid postcode') }
     end
   end
 
-  context 'validates supplier number format' do
+  context 'when validating supplier number' do
     let(:format_error) { ['Enter a valid LGFS supplier number'] }
 
-    it 'fails for incorrect format' do
-      allow(subject).to receive(:supplier_number).and_return('ABC123')
-      expect(subject).not_to be_valid
-      expect(subject.errors[:supplier_number]).to eq(format_error)
+    before do
+      allow(supplier).to receive(:supplier_number).and_return(supplier_number)
```

Comment:
Shouldn't really stub the thing under test. I think this is the equivalent (without stubbing):

```suggestion
      supplier.supplier_number = supplier_number
```

Reply:
changed